publicdb/status_display/status.py

```diff
@@ -0,0 +1,68 @@
+import datetime
+
+from publicdb.inforecords.models import Station, Pc
+from publicdb.histograms.models import Summary
+
+
+RECENT_NUM_DAYS = 7
+
+
+class StationStatus(object):
+
+    """Query the status of a station.
+
+    You can query the status of a single station, or get the status counts for
+    all stations.
+
+    """
+
+    def __init__(self):
+        yesterday = datetime.date.today() - datetime.timedelta(days=1)
+        recent_day = datetime.date.today() - datetime.timedelta(days=RECENT_NUM_DAYS)
+
+        self.stations = Station.objects.values_list('number', flat=True)
+        self.stations_with_current_data = Summary.objects.filter(date__exact=yesterday).values_list('station__number', flat=True)
```

Comment:
This does not ensure that that station had data, maybe it only has configs for that date.
Use the queryset manager methods defined by [SummaryQuerySet](https://github.com/HiSPARC/publicdb/blob/master/publicdb/histograms/models.py#L47)
```python
Summary.objects.with_data().filter(…
```
This also limits dates to the valid range, i.e. not greater than today, which sometimes occurs because of the GPS time issue.

Reply:
Yes, I was just now about to make that change, ;-).